App/test/app/UserProfileTest.java

```diff
@@ -0,0 +1,417 @@
+                int rowIndex = Bookings.getSelectedRow();
+                model = (DefaultTableModel) Bookings.getModel();
+                int bookingID = (int) model.getValueAt(rowIndex, 0);
+                java.sql.Date datein = (java.sql.Date) model.getValueAt(rowIndex,3);
+                java.sql.Date dateout= (java.sql.Date) model.getValueAt(rowIndex,4);
+                ResultSet rs = getResult("SELECT CURDATE()");
+                rs.next();
+                Date today = rs.getDate("CURDATE()");
+                if(getDateDifference(datein, today) < 3){
+                     JOptionPane.showMessageDialog(null, "You cannot modify now.", "WARNING!", JOptionPane.WARNING_MESSAGE);
+                     return;
+                }         
+                new ModifyBooking(username, bookingID).setVisible(true);
+                // this.dispose();
+            } 
+            catch(Exception e){
+            System.out.println(e);
+            }
+            
+            System.out.println("Modify Action Performed Test Successful");
+        }        
+    }
+    
+  
+
+    @Test
+    public void RatingOptionActionPerformedTest (){
+         System.out.println(" Testing RatingOptionActionPerformed()");    
+         
+         UserProfile frame ;
+         JButton RatingOption; 
+         int num = 10;
+         for (int i = 0; i < num; i++) {
+            // generate random username
+            username = generateAlphaNumeric(40);
+            frame = new UserProfile(username);
+            //get access to  button
+            RatingOption = (JButton) TestUtils.getChildNamed(frame, "RatingOption");
+            Bookings = (JTable)TestUtils.getChildNamed(frame, "Bookings");
+            JTextField Rating = (JTextField) TestUtils.getChildNamed(frame, "Rating");
+
+            assertNotNull("Rating Option inaccessible", RatingOption);
+            assertNotNull("Frame component inaccessible", frame);
+            
+            // to see if each UI component is accessible
+            System.out.println("Frame, Button found");
+
+            System.out.print("TEST CASE: " + i + "\n");
+            //randomised username 
+            System.out.println("USERNAME: " + username);
+            
+            // integrating random dates into datepicker 
+            Date dateIn = randDate();
+            Date dateOut = randDate();
+            frame.setVisible(true);
+
+            // testing if values of username were correct and implemented correctly in frame variable
+            assertEquals(username, frame.username);
+            
+            // testing the code of button functionality
+            try {
+                 // TODO add your handling code here:
+                int rowIndex = Bookings.getSelectedRow();
+                model = (DefaultTableModel) Bookings.getModel();
+                
+                int bookingID = (int) model.getValueAt(rowIndex, 0);
+                int rating = Integer.parseInt(Rating.getText());
```

Comment:
There seems to be an error in this bit

Reply:
Ow ow ow... okay I think I know the problem here.
![image](https://user-images.githubusercontent.com/32078894/120126681-40945080-c1e7-11eb-9194-92799c9ea2ee.png)

You did get selected row, but you didn't set the selected row first

Reply:
https://stackoverflow.com/questions/8661251/programmatically-select-a-row-in-jtable

Try set it this up,
table.setRowSelectionInterval(0, 0);